spec/index.md

```diff
@@ -3680,14 +3744,6 @@ if A.settings.controllers is not null:
 else:
   New_controllers = [M.caller]
 
-if New_compute_allocation > 0 or New_memory_allocation > 0:
```

Comment:
I recall that the freezing limit was not checked if the compute and memory allocation did not change. Now you specify it to be always checked. Did the implementation change in this regard?

Reply:
The implementation does not seem to have changed:
https://github.com/dfinity/ic/blob/b53c7fff38f7e5adc324283b162abd86bf356d4e/rs/execution_environment/src/execution/install_code.rs#L230

Reply:
I think that might be unintentional in the implementation and a side-effect of that helper being used both for canister creation and changing settings.

If we just look at canister creation, it makes sense to refuse creating canister that would immediately become frozen. I would suggest adjusting the implementation for that. Wdyt?

If you prefer to add the check here in spec, I can also do that.

Reply:
Thank you for adjusting the spec! I agree that creating a canister that is immediately frozen doesn't make much sense, but I'm not sure how relevant this is in practice (so I'm also fine if we keep the existing behavior now that we properly specify it).